k8s/k8s.go

```diff
@@ -0,0 +1,1291 @@
+package k8s
+
+import (
+	"sync"
+	"time"
+
+	snap_v1 "github.com/kubernetes-incubator/external-storage/snapshot/pkg/apis/crd/v1"
+	autopilotclientset "github.com/libopenstorage/autopilot-api/pkg/client/clientset/versioned"
+	storkclientset "github.com/libopenstorage/stork/pkg/client/clientset/versioned"
+	ocp_clientset "github.com/openshift/client-go/apps/clientset/versioned"
+	ocp_security_clientset "github.com/openshift/client-go/security/clientset/versioned"
+	"github.com/portworx/sched-ops/k8s/admissionregistration"
+	"github.com/portworx/sched-ops/k8s/apiextensions"
+	"github.com/portworx/sched-ops/k8s/apps"
+	"github.com/portworx/sched-ops/k8s/autopilot"
+	"github.com/portworx/sched-ops/k8s/batch"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/sched-ops/k8s/discovery"
+	"github.com/portworx/sched-ops/k8s/dynamic"
+	"github.com/portworx/sched-ops/k8s/externalstorage"
+	"github.com/portworx/sched-ops/k8s/openshift"
+	"github.com/portworx/sched-ops/k8s/operator"
+	"github.com/portworx/sched-ops/k8s/prometheus"
+	"github.com/portworx/sched-ops/k8s/rbac"
+	"github.com/portworx/sched-ops/k8s/storage"
+	"github.com/portworx/sched-ops/k8s/stork"
+	"github.com/portworx/sched-ops/k8s/talisman"
+	"github.com/sirupsen/logrus"
+	hook "k8s.io/api/admissionregistration/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	batch_v1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	rbac_v1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	storagev1beta1 "k8s.io/api/storage/v1beta1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/version"
+	dynamicclient "k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	masterLabelKey           = "node-role.kubernetes.io/master"
+	hostnameKey              = "kubernetes.io/hostname"
+	pvcStorageClassKey       = "volume.beta.kubernetes.io/storage-class"
+	pvcStorageProvisionerKey = "volume.beta.kubernetes.io/storage-provisioner"
+	labelUpdateMaxRetries    = 5
+)
+
+var deleteForegroundPolicy = meta_v1.DeletePropagationForeground
+
+// Ops is an interface to perform any kubernetes related operations
+type Ops interface {
+	NamespaceOps
```

Comment:
Instead of duplicating these individual ops, we should just compose this of the Ops that you extracted previously.
```
type Ops interface {
  core.Ops
  rbac.Ops
  ...
}
```
As the internal Ops have all of the below ops this list, we should directly be able to call, for instance `k8sOps.CreateServiceAccount()`